src/tests/topp/CMakeLists.txt

```diff
@@ -675,6 +667,25 @@ set_tests_properties("TOPP_FileMerger_10_out2" PROPERTIES DEPENDS "TOPP_FileMerg
 add_test("TOPP_FileMerger_10_out3" ${DIFF} -in1 FileMerger_10_trafo2.tmp -in2 ${DATA_DIR_TOPP}/FileMerger_10_trafo2.trafoXML)
 set_tests_properties("TOPP_FileMerger_10_out3" PROPERTIES DEPENDS "TOPP_FileMerger_10")
 
+#------------------------------------------------------------------------------
+# GNPSExport tests
+
+# one file using full_spectra:
+add_test("TOPP_GNPSExport_1" ${TOPP_BIN_PATH}/GNPSExport -in_cm ${DATA_DIR_TOPP}/GNPSExport_1_inp1.consensusXML -in_mzml ${DATA_DIR_TOPP}/GNPSExport_1_inp2.mzML -out GNPSExport_1_out.tmp -output_type full_spectra)
+add_test("TOPP_GNPSExport_1_out" ${DIFF} -in1 GNPSExport_1_out.tmp -in2 ${DATA_DIR_TOPP}/GNPSExport_1_out.mgf )
+
+# one file using merged_spectra:
+add_test("TOPP_GNPSExport_2" ${TOPP_BIN_PATH}/GNPSExport -in_cm ${DATA_DIR_TOPP}/GNPSExport_2_inp1.consensusXML -in_mzml ${DATA_DIR_TOPP}/GNPSExport_2_inp2.mzML -out GNPSExport_2_out.tmp -output_type merged_spectra)
+add_test("TOPP_GNPSExport_2_out" ${DIFF} -in1 GNPSExport_2_out.tmp -in2 ${DATA_DIR_TOPP}/GNPSExport_2_out.mgf)
+
+# two files using full_spectra:
+add_test("TOPP_GNPSExport_3" ${TOPP_BIN_PATH}/GNPSExport -in_cm ${DATA_DIR_TOPP}/GNPSExport_3_inp1.consensusXML -in_mzml ${DATA_DIR_TOPP}/GNPSExport_3_inp2.mzML ${DATA_DIR_TOPP}/GNPSExport_3_inp3.mzML -out GNPSExport_3_out.tmp -output_type full_spectra)
+add_test("TOPP_GNPSExport_3_out" ${DIFF} -in1 GNPSExport_3_out.tmp -in2 ${DATA_DIR_TOPP}/GNPSExport_3_out.mgf)
+
+# two files using merged_spectra:
+add_test("TOPP_GNPSExport_4" ${TOPP_BIN_PATH}/GNPSExport -in_cm ${DATA_DIR_TOPP}/GNPSExport_4_inp1.consensusXML -in_mzml ${DATA_DIR_TOPP}/GNPSExport_4_inp2.mzML ${DATA_DIR_TOPP}/GNPSExport_4_inp3.mzML -out GNPSExport_4_out.tmp)
+add_test("TOPP_GNPSExport_4_out" ${DIFF} -in1 GNPSExport_4_out.tmp -in2 ${DATA_DIR_TOPP}/GNPSExport_4_out.mgf)
```

Comment:
you will need to add dependencies:
```
set_tests_properties("TOPP_GNPSExport_4_out" PROPERTIES DEPENDS "TOPP_GNPSExport_4") 
```
for all your tests since they get executed in parallel and you need to *first* execute the tool before you can test its results